File: select_sort/select_sort.go
package main

import "fmt"

func SelectSort(arr []int) {
	var index int
	var temp int
	for i := 0; i < len(arr); i++ {
		index = i // index 记录无序区最小值索引
		for j := i + 1; j < len(arr); j++ {
			if arr[j] < arr[index] {
				index = j
			}
		}
		temp = arr[i]
		arr[i] = arr[index]
		arr[index] = temp
	}
}

func main() {
	arr := []int{2, 5, 3, 5, 6, 8, 1}
	SelectSort(arr)
	fmt.Println(arr)
}

File: select_sort/select_sort.c
#include <stdio.h>

void SelectSort(int *arr, int len)
{
    int temp, index;
    for (int i = 0; i < len; i++)
    {
        index = i;
        for (int j = i + 1; j < len; j++)
        {
            if (arr[j] < arr[index])
            {
                index = j;
            }
        }
        temp = arr[i];
        arr[i] = arr[index];
        arr[index] = temp;
    }
}

int main()
{
    int arr[] = {1, 3, 2, 5, 8, 3, 9, 4};
    SelectSort(arr, 8);
    for (int i = 0; i < 8; i++)
    {
        printf("%d ", arr[i]);
    }
    printf("\n");
    return 0;
}